requirements/constraints.txt

```diff
@@ -11,9 +11,11 @@
 # Common constraints for Open edX repos
 -c https://raw.githubusercontent.com/openedx/edx-lint/master/edx_lint/files/common_constraints.txt
 
-# Keep same platform version
-pylint<3.0
-pycodestyle<2.9.0
+# Versions >= 3.3.0 drop support for Python 3.8
+# Ensures a consistent pylint version across all Python environments to avoid triggering varying warnings
+pylint<3.3.0
+
+# Higher version has breaking changes in the platform as described here https://github.com/openedx/edx-platform/issues/35277
```

Comment:
I think we should start adding when this constraint should be dropped as part of this comment if the issue linked doesn't offer any other context. In this case, issue/35277 clearly says when the constraint is being unpinned, but when should we unpin pylint? I don't want this to be a blocker, but we need to offer more details when adding restrictions so we don't have to guess whether a constraint should be dropped.

Reply:
Actually, this is the format that's used in edx-platform:

```
# Date: 2020-02-26
# As it is not clarified what exact breaking changes will be introduced as per
# the next major release, ensure the installed version is within boundaries.
# Issue for unpinning: https://github.com/openedx/edx-platform/issues/35280
```
Should we consider adopting it?

Reply:
Sure, I'll add more info on the constraint and it probablya  good idea to have the same syntax across all repos.